src/Features/LanguageServer/Protocol/Handler/Completion/CompletionItemExtensions.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.Completion;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+
+    internal static class CompletionItemExtensions
+    {
+
+        // A list of the different property names that the insertion text of a completion item may be stored under.
+        private static readonly string[] InsertionTextPropertyNames = new string[] { "InsertionText", "InsertText" };
+
+        internal static string GetInsertionText(this CompletionItem item)
+        {
+            foreach (var insertionTextPropertyName in InsertionTextPropertyNames)
+            {
+                if (item.Properties.ContainsKey(insertionTextPropertyName))
+                {
+                    return item.Properties[insertionTextPropertyName];
```

Comment:
I'm not super familiar with roslyn completetion, tagging @heejaechang
Is the display text ever different from the insertion text?  It seems like we just set the insertion text to the display text - http://source.roslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs,368
but I might be wrong.

Reply:
We often set the insertion text differently than display text.   We only use the display text if there is no insertion text.